server/src/internalClusterTest/java/org/opensearch/index/codec/MultiCodecMergeIT.java

```diff
@@ -0,0 +1,172 @@
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertNoFailures;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class MultiCodecMergeIT extends OpenSearchIntegTestCase {
+
+    public void testForceMergeMultipleCodecs() throws ExecutionException, InterruptedException {
+
+        Map<String, String> codecMap = Map.of(
+            "best_compression",
+            "BEST_COMPRESSION",
+            "zstd_no_dict",
+            "ZSTD_NO_DICT",
+            "zstd",
+            "ZSTD",
+            "default",
+            "BEST_SPEED"
+        );
+
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            forceMergeMultipleCodecs(codec.getKey(), codec.getValue(), codecMap);
+        }
+
+    }
+
+    private void forceMergeMultipleCodecs(String finalCodec, String finalCodecMode, Map<String, String> codecMap) throws ExecutionException,
+        InterruptedException {
+
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        final String index = "test-index" + finalCodec;
+
+        // creating index
+        createIndex(
+            index,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+                .put("index.codec", "default")
+                .build()
+        );
+        ensureGreen(index);
+        Set<String> codecSet = new HashSet<>();
+
+        // ingesting and asserting segment codec mode for all four codecs
+        for (Map.Entry<String, String> codec : codecMap.entrySet()) {
+            ingestUsingCodec(index, codec.getKey(), codecSet);
+        }
+
+        assertTrue(codecSet.containsAll(codecMap.values()));
+
+        // force merge into final codec
+        ingestUsingCodec(index, finalCodec, codecSet);
+        final ForceMergeResponse forceMergeResponse = client().admin().indices().prepareForceMerge(index).setMaxNumSegments(1).get();
+
+        assertThat(forceMergeResponse.getFailedShards(), is(0));
+        assertThat(forceMergeResponse.getSuccessfulShards(), is(2));
+
+        flushAndRefreshIndex(index);
+
+        List<Segment> segments = getSegments(index).stream().filter(Segment::isSearch).collect(Collectors.toList());
+        assertEquals(1, segments.size());
+        assertTrue(segments.stream().findFirst().get().attributes.containsValue(finalCodecMode));
+    }
+
+    private void ingestUsingCodec(String index, String codec, Set<String> codecSet) throws InterruptedException, ExecutionException {
+
+        assertAcked(client().admin().indices().prepareClose(index));
+
+        assertAcked(
+            client().admin()
+                .indices()
+                .updateSettings(new UpdateSettingsRequest(index).settings(Settings.builder().put("index.codec", codec)))
+                .get()
+        );
+
+        assertAcked(client().admin().indices().prepareOpen(index));
```

Comment:
Van be moved to a separate method called ```changeCodec``` or ```useCodec``` and called from the current method and from L90

Reply:
changed